addons/laserpointer/XEH_postInit.sqf

```diff
@@ -7,32 +7,54 @@
 if (!hasInterface) exitWith {};
 
 GVAR(nearUnits) = [];
+GVAR(index) = -1;
+GVAR(laserClassesCache) = [] call CBA_fnc_createNamespace;
+GVAR(redLaserUnits) = [];
+GVAR(greenLaserUnits) = [];
 
 #include "initKeybinds.sqf"
 
 ["ace_settingsInitialized", {
-    //If not enabled, dont't add draw eventhandler or PFEH (for performance)
-    if (!GVAR(enabled)) exitWith {};
+    // if not enabled, dont't add draw eventhandler or PFEH (for performance)
+    if !(GVAR(enabled)) exitWith {};
 
-    // @todo. Maybe move to common?
     [{
-        // handle RHS / bugged vehicle slots
-        private _camPosAGL = positionCameraToWorld [0,0,0];
-        if !((_camPosAGL select 0) isEqualType 0) exitWith {};
-
-        private _nearUnits = [];
-        {
-            _nearUnits append crew _x;
-            if (count _nearUnits > 10) exitWith {
-                _nearUnits resize 10;
+        GVAR(nearUnits) = call FUNC(getNearUnits);
+    }, 5, []] call CBA_fnc_addPerFrameHandler;
+
+    // custom scheduler
+    [{
+        GVAR(index) = GVAR(index) + 1;
+        private _unit = GVAR(nearUnits) param [GVAR(index), objNull];
+
+        if (isNull _unit) exitWith {
+            GVAR(index) = -1;
+            GVAR(redLaserUnits) = [];
+            GVAR(greenLaserUnits) = [];
+        };
+
+        private _weapon = currentWeapon _unit;
+        private _laser = (_unit weaponAccessories _weapon) select 1;
+
+        if (_laser != "") then {
```

Comment:
I prefer exitWith to keep indentation a little lower. Just dropping my thoughts.